### PR TITLE
feat: add built-in function to encode a string as base64

### LINKED ISSUE #846: Support encoding of strings to base64
**Is your feature request related to a problem? Please describe.**

[Some APIs](https://docs.github.com/de/rest/repos/contents?apiVersion=2022-11-28#create-or-update-file-contents) require data to be sent as base64 encoded strings. Even for text values like textfile contents. It would therefore be handy to be able to encode strings right inside the feel engine via a OOTB function that supports strings.

**Describe the solution you'd like**

We should support an `encode base64` function that accepts a string as the input and returns a base64 encoded string as the output.

**Related issues**

https://github.com/camunda/connectors/issues/2023

### PR DESCRIPTION
**Pull request by sbuettner**
## Description

Add function to encode a string as base64

## Related issues

closes #846

### CODE REVIEW

File: src/main/scala/org/camunda/feel/impl/builtin/StringBuiltinFunctions.scala [hunk 1]
```diff
@@ -42,7 +43,8 @@ object StringBuiltinFunctions {
     "split"            -> List(splitFunction),
     "extract"          -> List(extractFunction),
     "trim"             -> List(trimFunction),
-    "uuid"             -> List(uuidFunction)
+    "uuid"             -> List(uuidFunction),
+    "encode base64"    -> List(encodeBase64Function)
```
Comment (saig0) [inline src/main/scala/org/camunda/feel/impl/builtin/StringBuiltinFunctions.scala:47]: :wrench: We could rename the function to `to base64()` to align with other data transformation functions, like `to json()` ([ref](https://github.com/camunda/feel-scala/issues/825)). 

What do you think about changing the name?

File: src/test/scala/org/camunda/feel/impl/builtin/BuiltinStringFunctionsTest.scala [hunk 2]
```diff
@@ -173,4 +173,11 @@ class BuiltinStringFunctionsTest extends AnyFlatSpec with Matchers with FeelInte
 
     eval(" string length(uuid()) ") should be(ValNumber(36))
   }
+
+  it should "return a string encoded as base64" in {
```
Comment (saig0) [inline src/test/scala/org/camunda/feel/impl/builtin/BuiltinStringFunctionsTest.scala:177]: :wrench: We need to group the test cases by name. By using the keyword `it`, it adds the test case to the previous group above. 

```suggestion
  "A encode base64() function" should "return a string encoded as base64" in {
```

File: src/test/scala/org/camunda/feel/impl/builtin/BuiltinStringFunctionsTest.scala [hunk 3]
```diff
@@ -173,4 +173,11 @@ class BuiltinStringFunctionsTest extends AnyFlatSpec with Matchers with FeelInte
...
+
+    eval(""" encode base64("FEEL") """) should be(ValString("RkVFTA=="))
+
+    eval(""" encode base64("Camunda") """) should be(ValString("Q2FtdW5kYQ=="))
```
Comment (saig0) [inline src/test/scala/org/camunda/feel/impl/builtin/BuiltinStringFunctionsTest.scala:181]: :wrench: We should add another test case that invokes the function with the named argument. This ensure that we don't change the parameter name accidentally. 

```
eval(""" encode base64(string: "Camunda") """) should be(ValString("Q2FtdW5kYQ=="))
```

**Review by saig0 — CHANGES_REQUESTED** [review]
@sbuettner thank you for your contribution. :tada: 

Looks very good. :+1: I have a few minor comments. Please have a look. :cookie:

File: src/main/scala/org/camunda/feel/impl/builtin/StringBuiltinFunctions.scala (hunk 1 above)
Comment (sbuettner) [reply src/main/scala/org/camunda/feel/impl/builtin/StringBuiltinFunctions.scala:47]: Yes as it was just a proposal and I wasnt happy with it either.

**Comment by sbuettner** [comment]
@saig0 Please take another look.

**Review by saig0 — APPROVED** [review]
@sbuettner LGTM. :rocket:

**Comment by saig0** [comment]
@sbuettner when do you need this feature? 

The next version 1.18.0 is planned for August/September. However, we could build the version earlier if needed.

**Comment by sbuettner** [comment]
Thanks for asking @saig0. Earlier would be great as we depend on it for some document related Connectors.

**Pull request merged**